### ISSUE TITLE
Why panic happens  when i use Decoder.Decode(v interface{})  decode the "bad-utf8-at-end.toml"

### ISSUE THREAD

**Issue by lieinsun** [BurntSushi/toml#352]
when i decode the test file,  panic happend. 
expect to return an error, instead of panic?

```
func decodeBadToml() {
	file, err := os.Open("github.com/!burnt!sushi/toml@v1.0.0/internal/toml-test/tests/invalid/encoding/bad-utf8-at-end.toml")
	if err != nil {
		fmt.Println(err)
		return
	}
	var parsed interface{}
	decode, err := toml.NewDecoder(file).Decode(&parsed)
	if err != nil {
		fmt.Println(err)
		return
	}

	fmt.Printf("%+v", decode)
}
```

```
panic: runtime error: slice bounds out of range [238:237] [recovered]
	panic: runtime error: slice bounds out of range [238:237]

goroutine 1 [running]:
github.com/BurntSushi/toml.parse.func1()
	C:/Users/a/go/pkg/mod/github.com/!burnt!sushi/toml@v1.0.0/parse.go:33 +0x21a
panic({0x2421a80, 0xc0007fc9a8})
	C:/Program Files/Go/src/runtime/panic.go:1047 +0x262
github.com/BurntSushi/toml.(*lexer).current(0xc0000c0280)
	C:/Users/a/go/pkg/mod/github.com/!burnt!sushi/toml@v1.0.0/lex.go:115 +0x90
github.com/BurntSushi/toml.(*lexer).emit(0xc0000c0280, 0x6)
	C:/Users/a/go/pkg/mod/github.com/!burnt!sushi/toml@v1.0.0/lex.go:131 +0xa5
github.com/BurntSushi/toml.lexMultilineString(0xc0000c0280)
	C:/Users/a/go/pkg/mod/github.com/!burnt!sushi/toml@v1.0.0/lex.go:725 +0x194
github.com/BurntSushi/toml.(*lexer).nextItem(0xc0000c0280)
	C:/Users/a/go/pkg/mod/github.com/!burnt!sushi/toml@v1.0.0/lex.go:84 +0x173
github.com/BurntSushi/toml.(*parser).next(0xc0000c0300)
	C:/Users/a/go/pkg/mod/github.com/!burnt!sushi/toml@v1.0.0/parse.go:96 +0x78
github.com/BurntSushi/toml.(*parser).topLevel(0xc0000c0300, {0x12, {0xc000502600, 0x0}, {0x0, 0x0}, {0x5, 0xe7, 0x1}})
	C:/Users/a/go/pkg/mod/github.com/!burnt!sushi/toml@v1.0.0/parse.go:184 +0x8bc
github.com/BurntSushi/toml.parse({0xc000502600, 0xf2})
	C:/Users/a/go/pkg/mod/github.com/!burnt!sushi/toml@v1.0.0/parse.go:71 +0x62a
github.com/BurntSushi/toml.(*Decoder).Decode(0xc0008fa400, {0x2264120, 0xc0008fa3f0})
	C:/Users/a/go/pkg/mod/github.com/!burnt!sushi/toml@v1.0.0/decode.go:146 +0xa45
main.decodeBadToml()
	C:/GoWorkspace/trivyfix/main.go:91 +0x219
main.main()
	C:/GoWorkspace/trivyfix/main.go:20 +0x17

```

**Comment by lieinsun** [comment]
panic happened when using trivy to scan the image layer. if image contains some invalid toml,  the panic will be difficult to recover

**Comment by arp242** [comment]
This is a duplicate of #329; trivy will scan all TOML files, including an unfixed test case I added.

It's fixed in version 1.1.0 here; Trivy updated their dependency a few weeks ago: https://github.com/aquasecurity/trivy/issues/1372 – it may be in the latest release, but you'll have to ask them.
